### PR TITLE
Merged with cow-crazy and added more functionality with the input component

### PR DESCRIPTION
**Pull request by TravisMWise**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -38,27 +40,39 @@ const main = async () => {
 
   // Create physics objects
   // Physics(Velocity, angularVelocity, colliderRadius)
-  let asteroidPhysics = new Physics(new Vector3(0, 0, -30), new Vector3(0, 0, 0), 0);
-  let ufoPhysics = new Physics(new Vector3(0, 0, 0), new Vector3(0, -200, 0), 0);
+  let asteroidPhysics = new Physics(new Vector3(0, 0, 0), new Vector3(0, 0, 0), 0);
+  let ufoPhysics = new Physics(new Vector3(0, 0, UFO_START_SPEED), new Vector3(0, UFO_START_ROT, 0), 0);
```
Comment (Armintron) [inline src/index.js:44]: Perhaps make a `UFO` class that is a subclass to `GameObject` so you dont have this pattern of variables everywhere?

**Review by Armintron — APPROVED** [review]
Very cool! Works like a charm.

My only nit would be that asteroids should spawn relative to the ship and not the bounding box so that you dont actually "see" the box. However, I know thats not the premise of this PR

**Review by ctcuff — COMMENTED** [review]
This is fantastic, great work! Only suggestion: would It be possible to create one `Asteroid` class that takes either an `asteroid0` or `asteroid1` model as a constructor parameter? Seems like `Ast0` and `Ast1` have more or less the same logic

**Comment by TheIceCreamBear** [comment]
> This is fantastic, great work! Only suggestion: would It be possible to create one `Asteroid` class that takes either an `asteroid0` or `asteroid1` model as a constructor parameter? Seems like `Ast0` and `Ast1` have more or less the same logic

I would add on to this and say that it shouldn't even take the model in and rather take a boolean flag or a number determining which model to use (a number in the event that we add more asteroids in the future)

**Comment by TravisMWise** [comment]
> Very cool! Works like a charm.
> 
> My only nit would be that asteroids should spawn relative to the ship and not the bounding box so that you dont actually "see" the box. However, I know thats not the premise of this PR

I ended up increasing the size of the box and decreasing how far the ufo can go within the box. This can be controlled by using the shipBoxScalar value in game-box.js. I don't want to spawn objects relative to the ship when we are limiting the ship to a box. But I think we could completely remove the bounding box and spawn the asteroids relative to the ship so you can move anywhere if that's something we wanna do.

**Pull request merged**